src/core/mutatus.scala

```diff
@@ -87,19 +103,30 @@ object `package` {
         encoder: Encoder[T],
         dao: Dao[T],
         idField: IdField[T]
-    ): Ref[T] = new Ref[T](
-      svc.readWrite.put(buildEntity()).getKey
-    )
+    ): mutatus.Result[Ref[T]] = Result {
+      new Ref[T](
+        svc.readWrite.put(buildEntity()).getKey
+      )
```

Comment:
Will this produce one of Mutatus's exception types, or will it potentially create a `Surprise`?

Reply:
I think that `buildEntity` could be considered as safe operation. I don't think there is any case in which it may throw exception. I think that all possible errors would appears at compile time, e.q lack of `IdField` or lack of `Encoder[T]` 
Although `svc.readWrite.put` may result with `DatastoreException`, I'll add handling of this with `.extenuate{
      case exc: DatastoreException => DatabaseException(exc)
    }`